File: sdkwork-app-sdk-flutter/lib/src/api/ab.dart
import '../http/client.dart';
import '../models.dart';

class AbApi {
  final HttpClient _client;
  
  AbApi(this._client);

  /// 实验反馈
  Future<PlusApiResultVoid?> submitExperimentFeedback(String experimentId, ExperimentFeedbackForm body) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/${experimentId}/feedback'), body: body, contentType: 'application/json');
    return response is PlusApiResultVoid ? response : null;
  }

  /// 退出实验
  Future<PlusApiResultVoid?> exitExperiment(String experimentId, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/${experimentId}/exit'), params: params);
    return response is PlusApiResultVoid ? response : null;
  }

  /// 所有特性开关
  Future<PlusApiResultListFeatureFlagVO?> listFeatureFlags() async {
    final response = await _client.get(ApiPaths.appPath('/abtest/features'));
    return response is PlusApiResultListFeatureFlagVO ? response : null;
  }

  /// 批量特性开关
  Future<PlusApiResultMapStringFeatureFlagVO?> batchCheckFeatures(BatchFeatureCheckForm body) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/features'), body: body, contentType: 'application/json');
    return response is PlusApiResultMapStringFeatureFlagVO ? response : null;
  }

  /// 上报实验曝光
  Future<PlusApiResultVoid?> trackExperimentExposure(ExperimentExposureForm body) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/exposure'), body: body, contentType: 'application/json');
    return response is PlusApiResultVoid ? response : null;
  }

  /// 上报实验事件
  Future<PlusApiResultVoid?> trackExperimentEvent(ExperimentEventForm body) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/events'), body: body, contentType: 'application/json');
    return response is PlusApiResultVoid ? response : null;
  }

  /// 上报实验转化
  Future<PlusApiResultVoid?> trackExperimentConversion(ExperimentConversionForm body) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/conversion'), body: body, contentType: 'application/json');
    return response is PlusApiResultVoid ? response : null;
  }

  /// 批量获取分组
  Future<PlusApiResultMapStringExperimentAssignmentVO?> batchGetAssignments(BatchAssignmentForm body) async {
    final response = await _client.post(ApiPaths.appPath('/abtest/assignments'), body: body, contentType: 'application/json');
    return response is PlusApiResultMapStringExperimentAssignmentVO ? response : null;
  }

  /// 实验结果
  Future<PlusApiResultExperimentResultsVO?> getExperimentResults(String experimentId) async {
    final response = await _client.get(ApiPaths.appPath('/abtest/${experimentId}/results'));
    return response is PlusApiResultExperimentResultsVO ? response : null;
  }

  /// 实验报告
  Future<PlusApiResultExperimentReportVO?> getExperimentReport(String experimentId) async {
    final response = await _client.get(ApiPaths.appPath('/abtest/${experimentId}/report'));
    return response is PlusApiResultExperimentReportVO ? response : null;
  }

  /// UI配置
  Future<PlusApiResultUiConfigVO?> getUiConfig() async {
    final response = await _client.get(ApiPaths.appPath('/abtest/ui-config'));
    return response is PlusApiResultUiConfigVO ? response : null;
  }

  /// 灰度配置
  Future<PlusApiResultListRolloutConfigVO?> listRolloutConfigs() async {
    final response = await _client.get(ApiPaths.appPath('/abtest/rollouts'));
    return response is PlusApiResultListRolloutConfigVO ? response : null;
  }

  /// 灰度发布
  Future<PlusApiResultRolloutCheckVO?> checkRollout(String rolloutKey) async {
    final response = await _client.get(ApiPaths.appPath('/abtest/rollout/${rolloutKey}'));
    return response is PlusApiResultRolloutCheckVO ? response : null;
  }

  /// 特性开关
  Future<PlusApiResultFeatureFlagVO?> checkFeatureFlag(String featureKey) async {
    final response = await _client.get(ApiPaths.appPath('/abtest/features/${featureKey}'));
    return response is PlusApiResultFeatureFlagVO ? response : null;
  }

  /// 可用实验
  Future<PlusApiResultListExperimentVO?> listAvailableExperiments() async {
    final response = await _client.get(ApiPaths.appPath('/abtest/experiments'));
    return response is PlusApiResultListExperimentVO ? response : null;
  }

  /// 实验详情
  Future<PlusApiResultExperimentDetailVO?> getExperimentDetail(String experimentId) async {
    final response = await _client.get(ApiPaths.appPath('/abtest/experiments/${experimentId}'));
    return response is PlusApiResultExperimentDetailVO ? response : null;
  }

  /// 个性化配置
  Future<PlusApiResultMapStringObject?> getPersonalizedConfig() async {
    final response = await _client.get(ApiPaths.appPath('/abtest/config'));
    return response is PlusApiResultMapStringObject ? response : null;
  }

  /// 获取实验分组
  Future<PlusApiResultExperimentAssignmentVO?> getExperimentAssignment(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.appPath('/abtest/assignment'), params: params);
    return response is PlusApiResultExperimentAssignmentVO ? response : null;
  }

  /// 算法配置
  Future<PlusApiResultAlgorithmConfigVO?> getAlgorithmConfig() async {
    final response = await _client.get(ApiPaths.appPath('/abtest/algorithm-config'));
    return response is PlusApiResultAlgorithmConfigVO ? response : null;
  }
}

File: sdkwork-app-sdk-python/sdkwork_app_sdk/api/ab.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import BatchAssignmentForm, BatchFeatureCheckForm, ExperimentConversionForm, ExperimentEventForm, ExperimentExposureForm, ExperimentFeedbackForm, PlusApiResultAlgorithmConfigVO, PlusApiResultExperimentAssignmentVO, PlusApiResultExperimentDetailVO, PlusApiResultExperimentReportVO, PlusApiResultExperimentResultsVO, PlusApiResultFeatureFlagVO, PlusApiResultListExperimentVO, PlusApiResultListFeatureFlagVO, PlusApiResultListRolloutConfigVO, PlusApiResultMapStringExperimentAssignmentVO, PlusApiResultMapStringFeatureFlagVO, PlusApiResultMapStringObject, PlusApiResultRolloutCheckVO, PlusApiResultUiConfigVO, PlusApiResultVoid

class AbApi:
    """ab API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def submit_experiment_feedback(self, experimentId: str, body: ExperimentFeedbackForm) -> PlusApiResultVoid:
        """实验反馈"""
        return self._client.post(f"/app/v3/api/abtest/{experimentId}/feedback", json=body)

    def exit_experiment(self, experimentId: str, params: Optional[Dict[str, Any]] = None) -> PlusApiResultVoid:
        """退出实验"""
        return self._client.post(f"/app/v3/api/abtest/{experimentId}/exit", params=params)

    def list_feature_flags(self) -> PlusApiResultListFeatureFlagVO:
        """所有特性开关"""
        return self._client.get(f"/app/v3/api/abtest/features")

    def batch_check_features(self, body: BatchFeatureCheckForm) -> PlusApiResultMapStringFeatureFlagVO:
        """批量特性开关"""
        return self._client.post(f"/app/v3/api/abtest/features", json=body)

    def track_experiment_exposure(self, body: ExperimentExposureForm) -> PlusApiResultVoid:
        """上报实验曝光"""
        return self._client.post(f"/app/v3/api/abtest/exposure", json=body)

    def track_experiment_event(self, body: ExperimentEventForm) -> PlusApiResultVoid:
        """上报实验事件"""
        return self._client.post(f"/app/v3/api/abtest/events", json=body)

    def track_experiment_conversion(self, body: ExperimentConversionForm) -> PlusApiResultVoid:
        """上报实验转化"""
        return self._client.post(f"/app/v3/api/abtest/conversion", json=body)

    def batch_get_assignments(self, body: BatchAssignmentForm) -> PlusApiResultMapStringExperimentAssignmentVO:
        """批量获取分组"""
        return self._client.post(f"/app/v3/api/abtest/assignments", json=body)

    def get_experiment_results(self, experimentId: str) -> PlusApiResultExperimentResultsVO:
        """实验结果"""
        return self._client.get(f"/app/v3/api/abtest/{experimentId}/results")

    def get_experiment_report(self, experimentId: str) -> PlusApiResultExperimentReportVO:
        """实验报告"""
        return self._client.get(f"/app/v3/api/abtest/{experimentId}/report")

    def get_ui_config(self) -> PlusApiResultUiConfigVO:
        """UI配置"""
        return self._client.get(f"/app/v3/api/abtest/ui-config")

    def list_rollout_configs(self) -> PlusApiResultListRolloutConfigVO:
        """灰度配置"""
        return self._client.get(f"/app/v3/api/abtest/rollouts")

    def check_rollout(self, rolloutKey: str) -> PlusApiResultRolloutCheckVO:
        """灰度发布"""
        return self._client.get(f"/app/v3/api/abtest/rollout/{rolloutKey}")

    def check_feature_flag(self, featureKey: str) -> PlusApiResultFeatureFlagVO:
        """特性开关"""
        return self._client.get(f"/app/v3/api/abtest/features/{featureKey}")

    def list_available_experiments(self) -> PlusApiResultListExperimentVO:
        """可用实验"""
        return self._client.get(f"/app/v3/api/abtest/experiments")

    def get_experiment_detail(self, experimentId: str) -> PlusApiResultExperimentDetailVO:
        """实验详情"""
        return self._client.get(f"/app/v3/api/abtest/experiments/{experimentId}")

    def get_personalized_config(self) -> PlusApiResultMapStringObject:
        """个性化配置"""
        return self._client.get(f"/app/v3/api/abtest/config")

    def get_experiment_assignment(self, params: Optional[Dict[str, Any]] = None) -> PlusApiResultExperimentAssignmentVO:
        """获取实验分组"""
        return self._client.get(f"/app/v3/api/abtest/assignment", params=params)

    def get_algorithm_config(self) -> PlusApiResultAlgorithmConfigVO:
        """算法配置"""
        return self._client.get(f"/app/v3/api/abtest/algorithm-config")
